selene-lib/src/rules/bad_string_escape.rs

```diff
@@ -0,0 +1,256 @@
+                    vec![
+                        "help: the maximum codepoint allowed in unicode escapes is `10ffff`"
+                            .to_owned(),
+                    ],
+                    Vec::new(),
+                ),
+                ReasonWhy::DoubleInSingle => Diagnostic::new(
+                    "bad_string_escape",
+                    "double quotes do not have to be escaped when inside single quoted strings"
+                        .to_owned(),
+                    Label::new(sequence.range.to_owned()),
+                ),
+                ReasonWhy::SingleInDouble => Diagnostic::new(
+                    "bad_string_escape",
+                    "single quotes do not have to be escaped when inside double quoted strings"
+                        .to_owned(),
+                    Label::new(sequence.range.to_owned()),
+                ),
+            })
+            .collect()
+    }
+
+    fn severity(&self) -> Severity {
+        Severity::Warning
+    }
+
+    fn rule_type(&self) -> RuleType {
+        RuleType::Correctness
+    }
+}
+
+struct BadStringEscapeVisitor {
+    sequences: Vec<StringEscapeSequence>,
+    roblox: bool,
+}
+
+struct StringEscapeSequence {
+    range: (usize, usize),
+    issue: ReasonWhy,
+}
+
+impl Visitor<'_> for BadStringEscapeVisitor {
+    fn visit_value(&mut self, node: &ast::Value) {
+        if_chain::if_chain! {
+            if let ast::Value::String(token) = node;
+            if let tokenizer::TokenType::StringLiteral {literal, multi_line, quote_type} = token.token_type();
+            if multi_line.is_none();
+            then {
+                let value_start = node.range().unwrap().0.bytes();
+
+                for captures in STRING_ESCAPE_REGEX.captures_iter(literal) {
+                    let start = value_start + captures.get(1).unwrap().start();
+
+                    match &captures[1] {
+                        "a" | "b" | "f" | "n" | "r" | "t" | "v" | "\\" => {},
+                        "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9" => {
+                            if &captures[2].len() > &1 {
+                                let hundreds = u16::from_str_radix(&captures[1], 10).unwrap() * 100;
+                                let tens = u16::from_str_radix(&captures[2][1..2], 10).unwrap();
+                                if hundreds+tens > 0xff {
+                                    self.sequences.push(
+                                        StringEscapeSequence{
+                                            range: (start, start+4),
+                                            issue: ReasonWhy::DecimalTooHigh,
+                                        }
+                                    );
+                                }
+                            }
+                        },
+                        "\"" => {
+                            if quote_type == &tokenizer::StringLiteralQuoteType::Single {
```

Comment:
I feel you should redefine `quote_type` to `let quote_type = *quote_type;`, then check directly.